### PR TITLE
Fix inconsistent timeout behaviour of check_ping plugin

### PR DESCRIPTION
**Pull request by bazzisoft**
Revert commit b7ee01a257d0bee30b3ef4d579ad56533f389e70 (Fix for issue #139) which incorrectly modifies timeout behaviour of the `check_ping` plugin.

See also:
https://github.com/nagios-plugins/nagios-plugins/issues/419
https://github.com/nagios-plugins/nagios-plugins/issues/323
https://github.com/nagios-plugins/nagios-plugins/issues/139

The above commit modified the `/bin/ping` command line to use the `-W` parameter instead of `-w` for specifying the timeout. `-w` is a global timeout for the command whereas `-W` specifies the maximum time to wait for a single ping response. This caused a number of issues:

1. For unresponsive hosts instead of the ping command timing out and providing useful output, viz:
`PING CRITICAL - Packet loss = 100%|rta=1200.000000ms;600.000000`
we get instead:
`CRITICAL - Plugin timed out`
This is because the `-W` param is effectively a timeout for the _last_ ping packet sent, so the ping command is executing for a max "timeout + num packets" seconds instead of "timeout" seconds. Thus the SIGALRM set to "timeout + 1" seconds is triggering and killing the plugin.

2. The above behaviour is inconsistent with other ping commands specified in `configure.ac` for other OS's such as BSD Unix. Eg: `-X` for `/bin/ping6` or `-t` for  `/bin/ping`, which both specify a global timeout like `-w` on GNU/Linux.

3. To properly address Issue #139 I believe we'll need to add an additional parameter to `check_ping` to specify the maximum wait time for a single ping packet.

### CODE REVIEW

**Comment by sawolf** [comment]
You bring up some good points here. I'll need to think on this for a bit.

**Comment by bazzisoft** [comment]
> You bring up some good points here. I'll need to think on this for a bit.

Sure, curious to hear what you come up with. I'm not sure the original problem specified in #139 still exists (i think it was exclusive to the old check_icmp?), but I can confirm that /bin/ping still sends 30 packets to an unresponsive host if the timeout is set to 30s. I don't think this is a problem though, it's how ping works internally by retrying failed packets, and it seems to correctly account for the extra packets when calculating the packet loss %.

**Comment by sawolf** [comment]
Sorry for the late response, but I did finally get back around to this. 

It looks like the "problem" reported in #139 (ICMP packets sent exceeds -p argument) was not the user's actual complaint (packet loss showed <100% even though host was completely dead). 

The user also claimed that they were having the same issue with both `check_icmp` and `check_ping`, and then that changing all checks to use `check_ping` fixed the issue. Given that I haven't reproduced the original problem yet, I suspect there was some other anomaly on their network - a dead host being brought up for a minute before crashing again, or something like that.

Anyways, #504 is clever, but simply reverting the change seems like the safer option.
